Source/Private/ASLayoutTransition.mm

```diff
@@ -156,18 +180,40 @@ - (void)calculateSubnodeOperationsIfNeeded
   ASLayout *pendingLayout = _pendingLayout->layout;
 
   if (previousLayout) {
+#if AS_IG_LIST_KIT
+    // IGListDiff completes in linear time O(m+n), so use it if we have it:
+    IGListIndexSetResult *result = IGListDiff(previousLayout.sublayouts, pendingLayout.sublayouts, IGListDiffEquality);
+    _insertedSubnodePositions = findNodesInLayoutAtIndexes(pendingLayout, result.inserts, &_insertedSubnodes);
+    _removedSubnodePositions = findNodesInLayoutAtIndexes(previousLayout, result.deletes, &_removedSubnodes);
+    for (IGListMoveIndex *move in result.moves) {
+      id subnode = previousLayout.sublayouts[static_cast<NSUInteger>(move.from)].layoutElement;
+      _subnodeMoves.push_back(std::pair<id, NSUInteger>(subnode, move.to));
```

Comment:
- A C-cast `((NSUInteger)move.from)` is probably sufficient instead of a `static_cast` (which is usually used for C++ types). Is any kind of cast actually necessary though – the implicit conversion shouldn't be dangerous.
- This is a good place to use `unowned id subnode` to avoid the extra retain/release pair that ARC will insert otherwise – we will retain the object into the `std::pair` on the next line so it's safe.

Reply:
The cast is entirely for "clarity", but I don't really know why I thought it helped :-| :D